discord/guild.py

```diff
@@ -1906,17 +1916,32 @@ async def edit(
 
             fields['system_channel_flags'] = system_channel_flags.value
 
+        features = set(self.features)
+
         if community is not MISSING:
-            features = []
             if community:
                 if 'rules_channel_id' in fields and 'public_updates_channel_id' in fields:
-                    features.append('COMMUNITY')
+                    features.add('COMMUNITY')
                 else:
                     raise ValueError(
                         'community field requires both rules_channel and public_updates_channel fields to be provided'
                     )
+            else:
+                features.discard('COMMUNITY')
+
+        if discoverable is not MISSING:
+            if discoverable:
+                features.add('DISCOVERABLE')
+            else:
+                features.discard('DISCOVERABLE')
+
+        if invites_disabled is not MISSING:
+            if invites_disabled:
+                features.add('INVITES_DISABLED')
+            else:
+                features.discard('INVITES_DISABLED')
 
-            fields['features'] = features
+        fields['features'] = list(features)
```

Comment:
Might be better to only pay for the cost of this cast if the features are set. As it stands right now everyone pays for both casts to `set` and back to `list`.